FaunaDB.Client/Types/Value.cs

```diff
@@ -81,7 +81,35 @@ public abstract class Value : Expr
             Field.At(indexes).Get(this).Match(
                 Success: value => value,
                 Failure: reason => NullV.Instance
-                );
+            );
+
+
+        /// <summary>
+        /// Attempts to coerce this value to given type T specified.
+        /// <para>
+        /// See <see cref="Decoder"/>
+        /// </para>
+        /// </summary>
+        /// <exception cref="System.InvalidOperationException">
+        /// Thrown if this instance of <see cref="Value"/> is not convertible to T.
+        /// Or if type T does not fullfil the requirements to be decoded.
+        /// </exception>
+        public T To<T>() =>
```

Comment:
I don't have a strong option about it. I was relying on exceptions the decoder throws. The functional counterpart that receives a codec as you said is fated to disappear, and if I not mistake we created `IResult` for this functional api to not return nulls or throw exceptions. Perhaps we can keep using `IResult` on version 1.0 and remove codecs on version 2.0, what you think?